### PR TITLE
MAINT: stats: remove use of interp2d from levy_stable._fitstart

### PR DESCRIPTION
**Pull request by ev-br**
#### Reference issue

A companion to gh-17180.

#### What does this implement/fix?

`levy_stable._fitstart` uses `interpolate.interp2d`,  which gh-17180 deprecates. Thus, replace interp2d usage by its direct analog, per https://gist.github.com/ev-br/8544371b40f414b7eaf3fe6217209bff

#### Additional information

Two commits here may probably be squashed: the first commit adds assertions that the new and old usage is indentical, and the second commit removes interp2d usage and assertsions.

### CODE REVIEW

**Comment by ev-br** [comment]
The red cross on CI seems to be a timeout, possibly an internal Azure glitch. Would appreciate if someone restarts it, I don't seem to have a restart button.

File: scipy/stats/_levy_stable/__init__.py [hunk 1]
```diff
@@ -531,6 +532,11 @@ def _fitstart_S1(data):
         [0, 0.064, 0.128, 0.191, 0.330, 0.478, 1.362],
         [0, 0.056, 0.112, 0.167, 0.285, 0.428, 1.274]]
 
+    j = np.argsort(nu_alpha_range)
```
Comment (mdhaber) [inline scipy/stats/_levy_stable/__init__.py:535]: Is this just to emphasize that the data has to be in order? It seems unnecessary, because the current tables _are_ in order, but I would understand if the intent is to future-proof the code.
(Since I'm not very familiar with `RectBivariateSpline` or `interp2d`, I am not a bit less confident about my understanding.)

Comment (ev-br) [reply scipy/stats/_levy_stable/__init__.py:535]: Yes. This stanza is straight from interp2d code. RectBivariateSpline errors out if the data is not sorted in the right order.

Comment (ev-br) [reply scipy/stats/_levy_stable/__init__.py:535]: Sorry, should not type before coffee. Let me try again: RectBivariateSpline requires the input in the ascending order, and errors otherwise. 
interp2d however argsorts the inputs. 
This all would unnecessary if the tables were in the ascending order, but they are not and I felt it's safer to add some uglyness then redo the tables.

Comment (mdhaber) [reply scipy/stats/_levy_stable/__init__.py:535]: Well, the first tables are already in order. They only need to be transposed. I wouldn't mind just adding `np.array` around the definition of the table itself and appending `.T` at the end. Then we wouldn't need the `.T` when you pass it into `RectBivariateSpline`. 

And the set of tables are based on `alpha_range` in reverse order, so the rows just need to be reversed before the transpose. Similarly, I wouldn't mind doing all that in the definition by packing the list into an array, reversing the rows and transposing `[::-1].T`.

I just think it would make the logical part of the code a little simpler if the tables were set up the way they are going to be used, and we can do that pretty simply without a full `argsort`. Adding a comment e.g. "Order with increasing `alpha_range` and transpose as required by `RectBivariateSpline`" might help clarify for future readers.

I tried it locally. Let me know if you'd like me to suggest it as a PR to your branch.

File: scipy/stats/_levy_stable/__init__.py [hunk 2]
```diff
@@ -575,32 +581,33 @@ def _fitstart_S1(data):
...
-    psi_1 = interpolate.interp2d(
-        nu_beta_range, nu_alpha_range, alpha_table, kind="linear"
-    )
+    j = np.argsort(alpha_range)
+    alpha_range_r = np.asarray(alpha_range)[j]
+    nu_c_table_r = np.asarray(nu_c_table)[j, :]
+    nu_zeta_table_r = np.asarray(nu_zeta_table)[j, :]
+
+    psi_1 = RectBivariateSpline(nu_beta_range, nu_alpha_range_r,
+                                alpha_table_r.T, kx=1, ky=1, s=0)
```
Comment (mdhaber) [inline scipy/stats/_levy_stable/__init__.py:590]:
```suggestion
                                alpha_table_r.T, kx=1, ky=1)
```
This is the default. Just a thought, unless you wanted to emphasize it.

File: scipy/stats/_levy_stable/__init__.py [hunk 3]
```diff
@@ -616,13 +623,14 @@ def phi_5_1(beta, alpha):
     nu_beta = (p95 + p05 - 2 * p50) / (p95 - p05)
 
     if nu_alpha >= 2.439:
-        alpha = np.clip(psi_1_1(nu_beta, nu_alpha)[0], np.finfo(float).eps, 2.)
-        beta = np.clip(psi_2_1(nu_beta, nu_alpha)[0], -1.0, 1.0)
+        eps = np.finfo(float).eps
+        alpha = np.clip(psi_1_1(nu_beta, nu_alpha)[0, 0], eps, 2.)
```
Comment (mdhaber) [inline scipy/stats/_levy_stable/__init__.py:627]: I guess `RectBivariateSpline` alows returns 2D output because it treats `x` and `y` as though they were each `1d` arrays, and it always interpolates on a full grid defined by the cartesian product of `x` and `y` (rather than, e.g. `zip`ing them)? 
Coming from `stats`, I would have expected the output to be 0D if `x` and `y` are 0D, 1D if one (or both) of `x` and `y` were `1D`, and 2D if either of `x` or `y` were 2D. That is, the output shape would be the shape you'd get broadcasting `x` against `y`.

**Review by mdhaber — COMMENTED** [review]
Looks like an identical refactor with no change in functionality (as expected). Just a few thoughts and questions about how to write it.

File: scipy/stats/_levy_stable/__init__.py (hunk 2 above)
Comment (ev-br) [reply scipy/stats/_levy_stable/__init__.py:590]: Yeah, it's inconsistent across various FITPACK wrappers (e.g. SmoothBivariateSpline has a differet default), so better be explicit and save a trip to the docs and then to the source to check that the docs are not stale.

File: scipy/stats/_levy_stable/__init__.py (hunk 3 above)
Comment (ev-br) [reply scipy/stats/_levy_stable/__init__.py:627]: Here's an analysis by Warren: https://github.com/scipy/scipy/issues/16396#issuecomment-1154175268. My take is that it's a mess :-).

**Comment by ev-br** [comment]
> I tried it locally. Let me know if you'd like me to suggest it as a PR to your branch.

Please do!

File: scipy/stats/_levy_stable/__init__.py (hunk 1 above)
Comment (ev-br) [reply scipy/stats/_levy_stable/__init__.py:535]: Yes, I was lazy and simply copied what interp2d was doing, with argsorts and transposes and all.  Please feel free to refactor however you deem better!

File: scipy/stats/_levy_stable/__init__.py (hunk 3 above)
Comment (mdhaber) [reply scipy/stats/_levy_stable/__init__.py:627]: OK. I guess my thought is that it would be nice if the recommended interface were to follow the usual broadcasting rules if we're going to eliminate `interp2d`. But it's certainly better than it could be (e.g. always squeezing).

**Comment by mdhaber** [comment]
Thanks @ev-br!

**Comment by ev-br** [comment]
Thank you Matt, it does look much better with your refactor!
